### ISSUE TITLE
No location coordinates?

### ISSUE THREAD

**Issue by MikeTannenbaum** [gcobb321/icloud3#63]
I'm getting this error (which may be caused by the lack of authentication but I was able to get the location info with an earlier RC before upgrading to 11)

```
Logger: custom_components.icloud3.device_tracker
Source: custom_components/icloud3/device_tracker.py:7934
Integration: icloud3 (documentation)
First occurred: 9:59:05 PM (2 occurrences)
Last logged: 9:59:05 PM

iCloud3 Error > ('Mikeyshassio', '') (mikeys_iphone_hassio), mikeys_iphone_hassio > iOS App Entity device_tracker.mikeys_iphone_hassio does not contain location attributes (latitude, longitude). iCloud3 will be restarted in the 10 minutes if this error continues.. -------------------------. Attributus-not_set, {}
iCloud3 Error > ('Arielleshassio', '') (arielles_iphone_hassio), arielles_iphone_hassio > iOS App Entity device_tracker.arielles_iphone_hassio does not contain location attributes (latitude, longitude). iCloud3 will be restarted in the 10 minutes if this error continues.. -------------------------. Attributus-not_set, {}
```

Thanks for all your work on this! Very excited to get this working.

//////
EDIT: Removed my first issue of icloud authentication...was user error, as so often is the case. I didn't understand there were two codes to input into HA: the "trusted device" code which served the ID right next to the box to type it in, and then the actual apple 2FA code.

**Comment by MikeTannenbaum** [comment]
OK I think I resolved this by taking care of the icloud authorization and some naming errors, but now I'm getting these which are leaving me clueless:

```
Logger: custom_components.icloud3.device_tracker
Source: custom_components/icloud3/device_tracker.py:7995
Integration: icloud3 (documentation)
First occurred: 10:13:04 PM (1 occurrences)
Last logged: 10:13:04 PM

►INTERNAL ERROR-RETRYING (_update_device_icloud:OverallUpdate-'<' not supported between instances of 'NoneType' and 'int')
```

```
Logger: custom_components.icloud3.device_tracker
Source: custom_components/icloud3/device_tracker.py:8032
Integration: icloud3 (documentation)
First occurred: 10:13:04 PM (1 occurrences)
Last logged: 10:13:04 PM

'<' not supported between instances of 'NoneType' and 'int'
Traceback (most recent call last):
  File "/config/custom_components/icloud3/device_tracker.py", line 2505, in _update_device_icloud
    event_msg = (f"Discarding > {info_msg} > GPS-{format_gps(latitude, longitude, gps_accuracy)}, "
  File "/config/custom_components/icloud3/device_tracker.py", line 8032, in _secs_to_time_str
    if secs < 60:
TypeError: '<' not supported between instances of 'NoneType' and 'int'
```

and finally this, which I'm not sure if it's related to icloud3 but in case it is...
```
Logger: homeassistant.components.device_tracker
Source: components/device_tracker/legacy.py:184
Integration: Device tracker (documentation, issues)
First occurred: 10:12:55 PM (1386 occurrences)
Last logged: 10:15:08 PM

The see service is not supported for this entity device_tracker.arielles_iphone_hassio
The see service is not supported for this entity device_tracker.mikeys_iphone_hassio
```

**Comment by gcobb321** [comment]
The error was caused by no data being returned from iCloud when iCloud3 was generating an error message. I took care of the error msg but I would check your devicenames and also see if there are any errors in the Event Log when iCloud3 starts. The devicename in the track_devices parameter must be the same on your phone’s Setting App > General > About Name field. I would think you have ‘Mikeys iPhone’. That means the track_devices parameter would be ‘mikeys_iphone > ...’ not ‘mikeys_iohone_hassio > ...’
